tests/CommandLine.Tests/Unit/Issue862Tests.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using FluentAssertions;
+using Xunit;
+
+namespace CommandLine.Tests.Unit
+{
+    #if NET7_0_OR_GREATER
+    public class Issue862Tests
+    {
+
+        [Fact]
+        public void Properties_with_required_modifier_are_required()
+        {
+            var arguments = Array.Empty<string>();
+            var result = new Parser().ParseArguments<Options>(arguments);
+
+            result.Errors.Should().ContainSingle(e => e.Tag == ErrorType.MissingRequiredOptionError);
+            result.Tag.Should().Be(ParserResultType.NotParsed);
+        }
+
+        private class Options
+        {
+            [Option("cols")]
+            public required int Cols { get; set; }

```

Comment:
What if you have a `required` modifier and `[Option(Required = false)]`? I would say, it should throw because of conflicting settings. Please add test for such case

Reply:
This is not a trivia 🤔
So I move the discussion on #862 thread, [here](https://github.com/commandlineparser/commandline/issues/862#issuecomment-1347988738)